ops/model.py

```diff
@@ -3596,3 +3619,56 @@ def _ensure_loaded(self):
         self._notice = self._container.get_notice(self.id)
         assert self._notice.type == self.type
         assert self._notice.key == self.key
+
+
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    """CloudCredential contains a cloud credential possibly with secrets redacted.
+
+    Used as the type of attribute `credential` in :class:`CloudSpec`.
+    """
+
+    auth_type: str
+    attributes: Dict[str, str]
+    redacted: List[str]
```

Comment:
We would normally put a docstring after each of these lines (which shows up in the generated API docs, and elsewhere).